packages/cosmic-swingset/test/test-make.js

```diff
@@ -1,26 +1,51 @@
+/* eslint-disable no-await-in-loop */
+/* global setTimeout */
 import test from 'ava';
 import { spawn } from 'child_process';
 import path from 'path';
+import fs from 'fs';
 
 const filename = new URL(import.meta.url).pathname;
 const dirname = path.dirname(filename);
 
-test('make and exec', async t => {
+const runMake = (
+  args,
+  stdio = ['ignore', 'ignore', 'inherit'],
+  cwd = `${dirname}/..`,
+) =>
+  new Promise((resolve, reject) =>
+    spawn('make', args, {
+      cwd,
+      stdio,
+    }).addListener('exit', code => {
+      if (code !== 0) {
+        reject(Error(`exit ${code} from: make ${args}`));
+        return;
+      }
+      resolve(0);
+    }),
+  );
+
+test.before(async _t => {
+  await runMake(['scenario2-setup']);
+});
+
+test.serial('make and exec', async t => {
   await new Promise(resolve =>
-    spawn('make', ['scenario2-setup'], {
+    spawn('bin/ag-chain-cosmos', {
       cwd: `${dirname}/..`,
       stdio: ['ignore', 'ignore', 'inherit'],
     }).addListener('exit', code => {
-      t.is(code, 0, 'make scenario2-setup exits successfully');
+      t.is(code, 0, 'exec exits successfully');
       resolve();
     }),
   );
   await new Promise(resolve =>
-    spawn('bin/ag-chain-cosmos', {
+    spawn('make', ['scenario2-run-chain-to-halt'], {
```

Comment:
I considered factoring `runMake` out of all the places where it could/should be used, but I thought it would be good to make it clear that much of the code isn't changing at all. I'll give it another look.

Reply:
as a separate commit that would be clear enough imo